### ISSUE TITLE
Processor crash when submitting proof to alfajores

### ISSUE THREAD

**Issue by prestwich** [celo-org/optics-monorepo#429]
This is consistently happening every time the processor syncs to this index

cause is a solidity revert with no message, which usually means an abi decoding error

```
fields | {"error":"(code: -32000, message: execution reverted, data: None)"}
-- | --
  |   | level | "ERROR"
  |   | span | {"message":"CommittedMessage { leaf_index: 151, message: OpticsMessage { origin: 3000, sender: 0x000000000000000000000000ba889d87b9aac1ac322e9ffc01040bdd16681e42, sequence: 2, destination: 1000, recipient: 0xabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcd, body: [115, 116, 97, 116, 105, 99, 32, 109, 101, 115, 115, 97, 103, 101] } }","proof":"Proof { leaf: 0x5d0a69475353468409d3cee744455271ffcb10bf47302f6854d344c6acea7f58, index: 151, path: [0x8d85054aa59756f42b43a4a6282e298fe8ec1a2d533b4753dae5c9e090338662, 0x5754ba7f3714edd41b4d25f50211cb15cebf19b9d59127f69f1112a086940f03, 0x00238aea801c72b99fbc5d1f76c3492dac9456ebdab7988cb28e0aa7a0eae5b0, 0x21ddb9a356815c3fac1026b6dec5df3124afbadb485c9ba5a3e3398a04b7ba85, 0xb071a4bd940588fb0a29a2102cd6ac68250ca7ef044427d973fd40628db94785, 0x0eb01ebfc9ed27500cd4dfc979272d1f0913cc9f66540d7e8005811109e1cf2d, 0x887c22bd8750d34016ac3c66b5ff102dacdd73f6b014e710b51e8022af9a1968, 0x4e4785d99fd75e2be7366928d63d4c134e579aefa515866b0345dd667242d885, 0x9867cc5f7f196b93bae1e27e6320742445d290f2263827498b54fec539f756af, 0xcefad4e508c098b9a7e1d8feb19955fb02ba9675585078710969d3440f5054e0, 0xf9dc3e7fe016e050eff260334f18a5d4fe391d82092319f5964f2e2eb7c1c3a5, 0xf8b13a49e282f609c317a833fb8d976d11517c571d1221a265d25af778ecf892, 0x3490c6ceeb450aecdc82e28293031d10c7d73bf85e57bf041a97360aa2c5d99c, 0xc1df82d9c4b87413eae2ef048f94b4d3554cea73d92b0f7af96e0271c691e2bb, 0x5c67add7c6caf302256adedf7ab114da0acfe870d449a3a489f781d659e8becc, 0xda7bce9f4e8618b6bd2f4132ce798cdc7a60e7e1460a7299e3c6342a579626d2, 0x2733e50f526ec2fa19a22b31e8ed50f23cd1fdf94c9154ed3a7609a2f1ff981f, 0xe1d3b5c807b281e4683cc6d6315cf95b9ade8641defcb32372f1c126e398ef7a, 0x5a2dce0a8a7f68bb74560f8f71837c2c2ebbcbf7fffb42ae1896f13f7c7479a0, 0xb46a28b6f55540f89444f63de0378e3d121be09e06cc9ded1c20e65876d36aa0, 0xc65e9645644786b620e2dd2ad648ddfcbf4a7e5b1a3a4ecfe7f64667a3f0b7e2, 0xf4418588ed35a2458cffeb39b93d26f18d2ab13bdce6aee58e7b99359ec2dfd9, 0x5a9c16dc00d6ef18b7933a6f8dc65ccb55667138776f7dea101070dc8796e377, 0x4df84f40ae0c8229d0d6069e5c8f39a7c299677a09d367fc7b05e3bc380ee652, 0xcdc72595f74c7b1043d0e1ffbab734648c838dfb0527d971b602bc216c9619ef, 0x0abf5ac974a1ed57f4050aa510dd9c74f508277b39d7973bb2dfccc5eeb0618d, 0xb8cd74046ff337f0a7bf2c8e03e10f642c1886798d71806ab1e888d9e5ee87d0, 0x838c5655cb21c6cb83313b5a631175dff4963772cce9108188b34ac87c81c41e, 0x662ee4dd2dd7b2bc707961b1e646c4047669dcb6584f0d8d770daf5d7e7deb2e, 0x388ab20e2573d171a88108e79d820e98f26c0b84aa8b2f4aa4968dbb818ea322, 0x93237c50ba75ee485f4c22adf2f741400bdf8d6a9cc7df7ecae576221665d735, 0x8448818bb4ae4562849e949e17ac16e0be16688e156b5cf15e098c627c0056a9] }","self":"ReplicaProcessor { interval: 10, replica: Ethereum(EthereumReplica { contract: EthereumReplicaInternal(0x7c478a7c2952d5827e39d93319906e8044e884b7), domain: 1000, name: \"alfajores\" }), home: Ethereum(EthereumHome { contract: EthereumHomeInternal(0xca445ba4554cb9f2ca90f63d14a73d568de67cc6), domain: 3000, name: \"kovan\" }), db: RocksDB { path: \"/app/db_path\" } }","name":"process"}
  |   | spans | [{"self":"ReplicaProcessor { interval: 10, replica: Ethereum(EthereumReplica { contract: EthereumReplicaInternal(0x7c478a7c2952d5827e39d93319906e8044e884b7), domain: 1000, name: \"alfajores\" }), home: Ethereum(EthereumHome { contract: EthereumHomeInternal(0xca445ba4554cb9f2ca90f63d14a73d568de67cc6), domain: 3000, name: \"kovan\" }), db: RocksDB { path: \"/app/db_path\" } }","name":"spawn"},{"message":"CommittedMessage { leaf_index: 151, message: OpticsMessage { origin: 3000, sender: 0x000000000000000000000000ba889d87b9aac1ac322e9ffc01040bdd16681e42, sequence: 2, destination: 1000, recipient: 0xabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcdabcd, body: [115, 116, 97, 116, 105, 99, 32, 109, 101, 115, 115, 97, 103, 101] } }","proof":"Proof { leaf: 0x5d0a69475353468409d3cee744455271ffcb10bf47302f6854d344c6acea7f58, index: 151, path: [0x8d85054aa59756f42b43a4a6282e298fe8ec1a2d533b4753dae5c9e090338662, 0x5754ba7f3714edd41b4d25f50211cb15cebf19b9d59127f69f1112a086940f03, 0x00238aea801c72b99fbc5d1f76c3492dac9456ebdab7988cb28e0aa7a0eae5b0, 0x21ddb9a356815c3fac1026b6dec5df3124afbadb485c9ba5a3e3398a04b7ba85, 0xb071a4bd940588fb0a29a2102cd6ac68250ca7ef044427d973fd40628db94785, 0x0eb01ebfc9ed27500cd4dfc979272d1f0913cc9f66540d7e8005811109e1cf2d, 0x887c22bd8750d34016ac3c66b5ff102dacdd73f6b014e710b51e8022af9a1968, 0x4e4785d99fd75e2be7366928d63d4c134e579aefa515866b0345dd667242d885, 0x9867cc5f7f196b93bae1e27e6320742445d290f2263827498b54fec539f756af, 0xcefad4e508c098b9a7e1d8feb19955fb02ba9675585078710969d3440f5054e0, 0xf9dc3e7fe016e050eff260334f18a5d4fe391d82092319f5964f2e2eb7c1c3a5, 0xf8b13a49e282f609c317a833fb8d976d11517c571d1221a265d25af778ecf892, 0x3490c6ceeb450aecdc82e28293031d10c7d73bf85e57bf041a97360aa2c5d99c, 0xc1df82d9c4b87413eae2ef048f94b4d3554cea73d92b0f7af96e0271c691e2bb, 0x5c67add7c6caf302256adedf7ab114da0acfe870d449a3a489f781d659e8becc, 0xda7bce9f4e8618b6bd2f4132ce798cdc7a60e7e1460a7299e3c6342a579626d2, 0x2733e50f526ec2fa19a22b31e8ed50f23cd1fdf94c9154ed3a7609a2f1ff981f, 0xe1d3b5c807b281e4683cc6d6315cf95b9ade8641defcb32372f1c126e398ef7a, 0x5a2dce0a8a7f68bb74560f8f71837c2c2ebbcbf7fffb42ae1896f13f7c7479a0, 0xb46a28b6f55540f89444f63de0378e3d121be09e06cc9ded1c20e65876d36aa0, 0xc65e9645644786b620e2dd2ad648ddfcbf4a7e5b1a3a4ecfe7f64667a3f0b7e2, 0xf4418588ed35a2458cffeb39b93d26f18d2ab13bdce6aee58e7b99359ec2dfd9, 0x5a9c16dc00d6ef18b7933a6f8dc65ccb55667138776f7dea101070dc8796e377, 0x4df84f40ae0c8229d0d6069e5c8f39a7c299677a09d367fc7b05e3bc380ee652, 0xcdc72595f74c7b1043d0e1ffbab734648c838dfb0527d971b602bc216c9619ef, 0x0abf5ac974a1ed57f4050aa510dd9c74f508277b39d7973bb2dfccc5eeb0618d, 0xb8cd74046ff337f0a7bf2c8e03e10f642c1886798d71806ab1e888d9e5ee87d0, 0x838c5655cb21c6cb83313b5a631175dff4963772cce9108188b34ac87c81c41e, 0x662ee4dd2dd7b2bc707961b1e646c4047669dcb6584f0d8d770daf5d7e7deb2e, 0x388ab20e2573d171a88108e79d820e98f26c0b84aa8b2f4aa4968dbb818ea322, 0x93237c50ba75ee485f4c22adf2f741400bdf8d6a9cc7df7ecae576221665d735, 0x8448818bb4ae4562849e949e17ac16e0be16688e156b5cf15e098c627c0056a9] }","self":"ReplicaProcessor { interval: 10, replica: Ethereum(EthereumReplica { contract: EthereumReplicaInternal(0x7c478a7c2952d5827e39d93319906e8044e884b7), domain: 1000, name: \"alfajores\" }), home: Ethereum(EthereumHome { contract: EthereumHomeInternal(0xca445ba4554cb9f2ca90f63d14a73d568de67cc6), domain: 3000, name: \"kovan\" }), db: RocksDB { path: \"/app/db_path\" } }","name":"process"}]
  |   | target | "processor::processor"
  |   | timestamp | "Jun 30 15:43:46.010"
  |   | ts | 2021-06-30T15:43:46.011Z
  |   | tsNs | 1625067826011051683
```

**Comment by anna-carroll** [comment]
@yourbuddyconner can we log the `status` of the message within the `process` span?

**Comment by prestwich** [comment]
you can also look into doing this yourself :)

```
use tracing::info;

fn process(...) {
    ...
    info!("log message with {} and {}", param1, param2);
    ...
}
````

**Comment by prestwich** [comment]
if you want it on `process` I would suggest doing it on the process functions in `optics-base/src/replica.rs`

**Comment by prestwich** [comment]
This crash is likely an accidental deployment of outdated contract code. We should prioritize #388 and improved typechain generation to avoid this sort of thing

In the meantime, we've run a new deploy

**Comment by prestwich** [comment]
```


fields | {"error":"(code: -32016, message: The execution failed due to an exception., data: Some(String(\"Reverted\")))"}
-- | --
  |   | level | "ERROR"
  |   | span | {"replicas":"[\"kovan\"]","self":"Processor { interval: 10, prover: RwLock { mr: 536870911, s: Semaphore { permits: 536870911 }, c: UnsafeCell }, replica_tasks: RwLock { mr: 536870911, s: Semaphore { permits: 536870911 }, c: UnsafeCell }, core: AgentCore { home: Ethereum(EthereumHome { contract: EthereumHomeInternal(0x864f62a45954fbe451218e0f84b013e2e1edc3b1), domain: 1000, name: \"alfajores\" }), replicas: {\"kovan\": Ethereum(EthereumReplica { contract: EthereumReplicaInternal(0x2504511b3607dce7a3c11ea93d80bdeed9b5c717), domain: 3000, name: \"kovan\" })}, db: RocksDB { path: \"/app/db_path\" } } }","name":"run_many"}
  |   | spans | [{"self":"Processor { interval: 10, prover: RwLock { mr: 536870911, s: Semaphore { permits: 536870911 }, c: UnsafeCell }, replica_tasks: RwLock { mr: 536870911, s: Semaphore { permits: 536870911 }, c: UnsafeCell }, core: AgentCore { home: Ethereum(EthereumHome { contract: EthereumHomeInternal(0x864f62a45954fbe451218e0f84b013e2e1edc3b1), domain: 1000, name: \"alfajores\" }), replicas: {\"kovan\": Ethereum(EthereumReplica { contract: EthereumReplicaInternal(0x2504511b3607dce7a3c11ea93d80bdeed9b5c717), domain: 3000, name: \"kovan\" })}, db: RocksDB { path: \"/app/db_path\" } } }","name":"run_all"},{"replicas":"[\"kovan\"]","self":"Processor { interval: 10, prover: RwLock { mr: 536870911, s: Semaphore { permits: 536870911 }, c: UnsafeCell }, replica_tasks: RwLock { mr: 536870911, s: Semaphore { permits: 536870911 }, c: UnsafeCell }, core: AgentCore { home: Ethereum(EthereumHome { contract: EthereumHomeInternal(0x864f62a45954fbe451218e0f84b013e2e1edc3b1), domain: 1000, name: \"alfajores\" }), replicas: {\"kovan\": Ethereum(EthereumReplica { contract: EthereumReplicaInternal(0x2504511b3607dce7a3c11ea93d80bdeed9b5c717), domain: 3000, name: \"kovan\" })}, db: RocksDB { path: \"/app/db_path\" } } }","name":"run_many"}]
  |   | target | "processor::processor"
  |   | timestamp | "Jun 30 21:21:00.665"
  |   | ts | 2021-06-30T21:21:00.665Z
  |   | tsNs | 1625088060665817141


```

**Comment by prestwich** [comment]
this is a truely odd one. 

`Reverted 0x`

**Comment by prestwich** [comment]
should be fixed by #439

**Comment by anna-carroll** [comment]
Was fixed by #439 

Casting the `recipient` as a contract caused Solidity to throw under the hood when `recipient` was not, in fact, a contract; using low-level call fixes this 😄
